Userland/Utilities/shuf.cpp

```diff
@@ -1,58 +1,70 @@
 /*
  * Copyright (c) 2021, Ben Wiederhake <BenWiederhake.GitHub@gmx.de>
+ * Copyright (c) 2022, Eli Youngs <eli.m.youngs@gmail.com>
  *
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
 #include <AK/Random.h>
 #include <AK/String.h>
 #include <AK/Vector.h>
+#include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
 #include <LibMain/Main.h>
-#include <errno.h>
-#include <stdio.h>
 #include <stdlib.h>
-#include <unistd.h>
 
-ErrorOr<int> serenity_main([[maybe_unused]] Main::Arguments arguments)
+ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    TRY(Core::System::pledge("stdio"sv));
-
-    Vector<String> lines;
-
-    char* buffer = nullptr;
-    for (;;) {
-        size_t n = 0;
-        errno = 0;
-        ssize_t buflen = getline(&buffer, &n, stdin);
-        if (buflen == -1 && errno != 0) {
-            perror("getline");
-            exit(1);
+    TRY(Core::System::pledge("stdio rpath"));
+
+    Core::ArgsParser args_parser;
+    StringView path;
+    bool is_zero_terminated = false;
+
+    args_parser.add_positional_argument(path, "File", "file", Core::ArgsParser::Required::No);
+    args_parser.add_option(is_zero_terminated, "Split input on \\0, not newline", "zero-terminated", 'z');
+
+    args_parser.parse(arguments);
+
+    auto file = TRY(Core::Stream::File::open_file_or_standard_stream(path, Core::Stream::OpenMode::Read));
+    ByteBuffer buffer = TRY(file->read_all());
+
+    u8 input_delimiter = is_zero_terminated ? '\0' : '\n';
+    Vector<Bytes> lines;
```

Comment:
You might even be able to make these `ReadonlyBytes`

Reply:
If I just set this to ReadonlyBytes as is, the Fisher-Yates shuffle complains when I try to swap spans using a tmp variable. Let me know if there's an easy way around that and I'm happy to make this change.

Reply:
That's what I was afraid of, then Bytes is fine